src/features/live_sensor/atoms/sensorResource.ts

```diff
@@ -0,0 +1,55 @@
+import { apiClient } from '~core/apiClientInstance';
+import { createAtom } from '~utils/atoms';
+import { createAsyncAtom } from '~utils/atoms/createAsyncAtom';
+import { UPDATE_ENDPOINT_PATH } from '../constants';
+import { collectedPointsAtom } from './collectedPoints';
+import { sensorDataAtom } from './sensorData';
+import { resourceTriggerAtom } from './triggerResource';
+import type { FeatureCollection } from '~utils/geoJSON/helpers';
+
+const resourceDepsAtom = createAtom(
+  {
+    resourceTriggerAtom,
+  },
+  ({ get, getUnlistedState, schedule }, state: FeatureCollection | null = null) => {
+    const trigger = get('resourceTriggerAtom');
+    if (trigger === 0) return state;
+
+    const newState = getUnlistedState(collectedPointsAtom);
+    state = { ...newState };
+
+    if (state?.features.length) {
+      // schedule an effect to clear collected data after request was triggered
+      schedule((dispatch) => dispatch(sensorDataAtom.resetAllData()));
+    }
+    return state;
+  },
+);
```

Comment:
it seems to have nothing to do with resource dependencies, better to make a separate atom for this
```
 if (state?.features.length) {
      // schedule an effect to clear collected data after request was triggered
      schedule((dispatch) => dispatch(sensorDataAtom.resetAllData()));
    }
 ```
 
 In general you don't need a dependency atom here, you can pass null instead, and call the .request action in the required queries manually

Reply:
still need to trigger reset only after request ran